components.md

```diff
@@ -94,11 +94,18 @@ sequenceDiagram
 :::
 
 ## Coordinator
-A component primarily concerned with routing/coordinating the message flow between other Components.
+A component primarily tasked with routing/coordinating the message flow between other Components.
 It represents the intermediate zmq brokers, proxies or somesuch.
+
+Every network needs at least one Coordinator, as all other Components within a Node directly exchange messages only with the Coordinator.
+
+Every Node must have exactly one Coordinator.
```

Comment:
> Thus, even if we have only 1 Actor, 1 Director and 1 Observer, we might need 1 Coordinator for the control channel (heavily discussed with DEALER and ROUTER socket comms), 1 Coordinator for the data stream (rather simple zmq proxy with XPUB and XSUB sockets, to enable n-to-n publishing) 1 Coordinator for the logging stream (same thing as the data stream, but not the same entity, i.e. not the same ports at least), which are 3 Coordinators.

My assumption when writing this was that all 3 channels (control, data, logging) are coordinated by a Coordinator, which I did not see and disadvantages of. I guess we'll have to discuss that further (elsewhere).

Reply:
> for every Node (i.e. every machine involved)

I think we'll have to iterate on the formulation of "Node". The way I understand and intend is _not_ to restrict one Node to one machine (hence the "may" in "This may be a single application using one or more threads or processes."). A Node should be the "local" context in the sense that 1 Coordinator and N other Components make up a local "community"/graph, the Node (irrespective of location). 
We need to expand on the understanding of "local" here.

One wrinkle that this uncovers (thank you!) is that we want to enable a Node to use thread queues in LMT, which is of course not possible with its Components distributed between machines. 
We should be able to cover this with additional restrictions on LMT usage. Maybe key the restrictions on the individual capabilities of the various methods, and explicitly enumerate the methods.

Reply:
> all 3 channels (control, data, logging) are coordinated by a Coordinator, which I did not see and disadvantages of. 

It seems I have missed out on some other discussions (i.e. #32), I will have to read up on them to catch how new terms are used. Just for me, the term "Coordinator" was "one entity which helps to enable the communication", and multiple entities might be needed for successful working with multiple channels. It is no problem (for me) to stuff all needed Coordinators multithreaded in a single process for simplicity, but they are still separate things. 

Also, I thought the _whole deal_ of the LMT Node concept was to have it fully local, independent of zmq and network infrastructure, e.g. with queues. As far as I understood [queues](https://docs.python.org/3/library/queue.html) in python, they are all inside one process, which would be fine for that purpose. 

> A Node should be the "local" context in the sense that 1 Coordinator and N other Components make up a local "community"/graph, the Node (irrespective of location). 

Ok, with the additional behavior that one Node is not restricted to one machine, but serves as network delimiter/namespace or similar, this is fine for me, if we give one Node "one and only one control-channel Coordinator". I don't think we would like to have multiple data-channel Coordinators (proxies) by design at all, e.g. with one per Node, because one big deal of having this one proxy in one central place would be for Observers to only listen at one point. Like this we do not need to start with issues like multi-hop-like distribution of the data, as we are discussing things for the control-Coordinators. If we would ever need an additional data-channel proxy, this would be for performance issues, I think, and then I don't really think that we would get this kind of issue within the scope of our goals.

Reply:
> Also, I thought the whole deal of the LMT Node concept was to have it fully local, independent of zmq and network infrastructure, e.g. with [queues](https://docs.python.org/3/library/queue.html). As far as I understood queues in python, they are all inside one process, which would be fine for that purpose.

Initially yes, but then it turned out that zmq can also do the "local" stuff (ipc, inproc), so it seems good to add that to the LMT options. Maybe we should limit LMT to queues and zmq inproc, and if a node is in LMT, it consequently has to live in one process. What do you think?

Reply:
> I don't think we would like to have multiple data-channel Coordinators (proxies) by design at all, e.g. with one per Node, because one big deal of having this one proxy in one central place would be for Observers to only listen at one point. Like this we do not need to start with issues like multi-hop-like distribution of the data, as we are discussing things for the control-Coordinators.

Darn it, of course, I hadn't thought of that, yet. I think I agree.
We could make it so our "Coordinator" is purely about the control channel, and have another (singleton) Component be about data proxying (and maybe logging at the same time, as it's similar). I'll open an issue.